hedyorg/hedy
# 🖊️ Add user-friendly errors for calculations with incorrect types #3465

Fixes #3465
Add custom error message when calculations are passed incompatible types.

**How to test**
- All automated tests should pass
- Run Hedy locally and navigate to specified level. Run the following code and supply a string instead of a number. Ensure that there is a nice error displayed and no python error leaks:
Level 6
```
hoeken = ask 'Hoeveel hoeken krijgt dit figuur?'
hoek = 360 / hoeken
print hoek
```
Level 12
```
hoeken = ask 'Hoeveel hoeken krijgt dit figuur?'
hoek = 360 / hoeken
print hoek
```
Level 12. Note that addition should show a different error since concatenation works from level 12.
```
hoeken = ask 'Hoeveel hoeken krijgt dit figuur?'
hoek = 360 + hoeken
print hoek
```
